pkg/cmd/extension/manager.go

```diff
@@ -802,3 +825,23 @@ func codesignBinary(binPath string) error {
 	cmd := exec.Command(codesignExe, "--sign", "-", "--force", "--preserve-metadata=entitlements,requirements,flags,runtime", binPath)
 	return cmd.Run()
 }
+
+// cleanExtensionUpdateDir should be used before installing extensions to avoid past metadata creating problems.
+func (m *Manager) cleanExtensionUpdateDir(name string) error {
+	updatePath := m.UpdateDir(name)
+	if _, err := os.Stat(updatePath); err == nil {
+		if err := os.RemoveAll(updatePath); err != nil {
+			return fmt.Errorf("failed to remove previous extension update state: %w", err)
+		}
+	}
+
+	return nil
+}
```

Comment:
Might need to consider bubbling up the errors from `os.stat` due to something other than the file not existing.

Reply:
Just pushed changes, don't like the aesthetics of the change but they're done.